jobrunner/service.py

```diff
@@ -50,5 +55,16 @@ def sync_wrapper():
             time.sleep(sleep_after_error)
 
 
+def record_stats_wrapper():
+    """Wrap the record_stats call with logging context and an exception handler."""
+    while True:
+        try:
+            record_stats.main()
```

Comment:
What's the return for, OOI?

Reply:
Good point, have added comment:
```py
            # `main()` should loop forever, if it exits cleanly that means it
            # wasn't configured to run so we should exit the thread rather than
            # looping
```